Orange/widgets/widget.py

```diff
@@ -508,6 +508,14 @@ def closeContext(self):
         reinitializing the user interface (e.g. combo boxes) with the new
         data.
         """
+        # Clear schema-only settings when *closing* the context
+        # FIXME: Sadly, schema-only settings aren't cleared when non-contextual
+        # SettingsHandler is used
+        if self.current_context is not None:
```

Comment:
With this change, https://github.com/biolab/orange3/blob/master/Orange/widgets/visualize/owscatterplot.py#L581 is now redundant.